app/models/concerns/foreman_openscap/openscap_proxy_core_extensions.rb

```diff
@@ -4,18 +4,22 @@ module OpenscapProxyCoreExtensions
 
     included do
       validate :openscap_proxy_has_feature
+      validate :scap_client_class_present
+      after_save :update_scap_client
     end
 
-    def update_scap_client_params(proxy_id)
-      new_proxy = SmartProxy.find proxy_id
+    def update_scap_client
+      update_scap_client_params if openscap_proxy_id_changed? && openscap_proxy_id
```

Comment:
If you disassociate proxy, lookup values are not changed because you have puppetclasses for scap_client assigned. Note that user still has to assign classes to host manually (or make sure they are inherited from hostgroup). Ideally, user would just select/unselect openscap_proxy, and we would take care of assigning puppetclasses with correct lookup values. But the host form will display outdated/incorrect state of things with this approach. Therefore we have this "semi-manual" setup as a compromise.